### PR TITLE
feat: Auth & Profile Page

### PR DESCRIPTION
**Pull request by Vishruta-Patil**
- Authentication -
Sign in
Login 
 Logout

- Profile Page - 
  Profile page UI
 Fetch profile details
 Edit profile

- other - 
 Redux setup
Added Protected Routes
 Made mobile responsive

PS - Ignore commented code, need it for future reference

### CODE REVIEW

**Review by devananth — APPROVED** [review]
- Code looks clean, liked the way you have checked values and asssigned default values if not.
- Good work !

File: src/services/authService.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    name: string;
+    email: string;
+  }) => {
+    try {
+      // **** Add doc in collection without id ****
+      // await addDoc(collection(db, "users"), {
+      //   name,
+      //   username,
+      //   email,
+      // });
```
Comment (vaibhav18matere) [inline src/services/authService.ts:31]: this seems ambiguous. you can remove commented code, if not require for further ref.

**Review by vaibhav18matere — APPROVED** [review]
- clean and readable code.
- variable naming is very informative.
you can merge this

File: src/services/authService.ts (hunk 1 above)
Comment (Vishruta-Patil) [reply src/services/authService.ts:31]: Thanks , but I need it for my reference

**Pull request merged**